build.sh

```diff
@@ -1,37 +1,45 @@
 #!/bin/bash -e
 # This script is used to build, test and squash the OpenShift Docker images.
 #
-# Name of resulting image will be: 'NAMESPACE/BASE_IMAGE_NAME-VERSION-OS'.
+# Resulting image will be tagged: 'name:version' and 'name:latest'. Name and version
+#                                  are values of labels from resulted image
 #
-# BASE_IMAGE_NAME - Usually name of the main component within container.
-# OS - Specifies distribution - "rhel7" or "centos7"
+# OS - Specifies distribution - "rhel7", "centos7" or "fedora"
 # VERSION - Specifies the image version - (must match with subdirectory in repo)
 # TEST_MODE - If set, build a candidate image and test it
 # TAG_ON_SUCCESS - If set, tested image will be re-tagged as a non-candidate
 #       image, if the tests pass.
 # VERSIONS - Must be set to a list with possible versions (subdirectories)
-# OPENSHIFT_NAMESPACES - Which of available versions (subdirectories) should be
-#       put into openshift/ namespace.
 
 OS=${1-$OS}
 VERSION=${2-$VERSION}
 
 DOCKERFILE_PATH=""
 
-test -z "$BASE_IMAGE_NAME" && {
-  BASE_DIR_NAME=$(echo $(basename `pwd`) | sed -e 's/-[0-9]*$//g')
-  BASE_IMAGE_NAME="${BASE_DIR_NAME#s2i-}"
-}
-
 # Perform docker build but append the LABEL with GIT commit id at the end
 function docker_build_with_version {
   local dockerfile="$1"
+  echo "-> Version ${dir}: building image from '${dockerfile}' ..."
+
   git_version=$(git rev-parse --short HEAD)
   BUILD_OPTIONS+=" --label io.openshift.builder-version=\"${git_version}\""
   if [[ "${UPDATE_BASE}" == "1" ]]; then
     BUILD_OPTIONS+=" --pull=true"
   fi
-  docker build ${BUILD_OPTIONS} -t ${IMAGE_NAME} -f "${dockerfile}" .
+
+  docker build ${BUILD_OPTIONS} -f "${dockerfile}" . | tee build.log
+  local IMAGE_ID=$(cat build.log | awk '/Successfully built/{print $NF}')
+  rm build.log
```

Comment:
Could we use:
```bash
local docker_cmd=(the command)
{ IMAGE_ID=$("${docker_cmd[@]}" | tee /dev/fd/$fd | awk '/Successfully built/{print $NF}'); } {fd}>&1
```

Or something similar, to avod creating temporary "build.log" file?